### PR TITLE
Make tests runnable in vscode

### PR DESCRIPTION
**Pull request by vkresch**
Resolves https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/537

* **Analysis**: 
I added the path to the src in the __init__.py file so that we do not need to define the variable each time manually.
That way the visual studio code test can now also be executed and we can also now debug tests in vscode.

* **Implementation**: Mostly documentation update and path fixing.

* **Performed tests**:
Run all the tests in terminal and in vscode.

* **Documentation**: 
Updated the documentation und how to run the tests.

* **Check list**:

- [x] I extended the Github Actions CI test units with the corresponding tests for this new feature (if needed).
- [x] I extended the Sphinx documentation (if needed).
- [ ] I changed the README file (if needed).
- [ ] I extended the help section (if needed).
- [ ] I created a new issue if there is further work left to be done (if needed).

**Work effort**: vkresch/jdsika: 3h (already calculated)

### CODE REVIEW

File: docs/fsm/graphviz/draw_state_diagrams.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+import os
+import sys
+
+sys.path.insert(
+    0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../src"))
```
Comment (jdsika) [inline docs/fsm/graphviz/draw_state_diagrams.py:5]: You walked the 3 folders up before you walk them down again? I remember you saying something about that "being hackish"? :D

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../src")))
```
Comment (jdsika) [inline tests/__init__.py:4]: Does this make my [Setup](https://tezos-reward-distributor-organization.github.io/tezos-reward-distributor/contributors.html) section obsolete?

File: docs/testing.rst [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+Tests
+========
 
-The tests in TRD are run in the following order:
+Test Execution
+---------------
+To run the tests locally you need to have `pytest <https://pypi.org/project/pytest/>`_ installed:
+::
+
+    pip install pytest
```
Comment (jdsika) [inline docs/testing.rst:9]: Should we split the requirements.txt in two files where one is for users and the other for devs? Then add pytest there as well and install everything through the file? [See comment here already](https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/blob/master/requirements.txt)

File: docs/testing.rst [hunk 4]
```diff
@@ -57,4 +72,64 @@ Verification (WIP)
...
+
+
+Visual Studio Code Test Execution (optional)
+----------------------------------------------
+
+If you are using `Visual Studio Code <https://code.visualstudio.com/>`_ and want to use it with the 
+`Python Test Explorer <https://marketplace.visualstudio.com/items?itemName=LittleFoxTeam.vscode-python-test-adapter>`_ 
+extension you need to first download it at the Extension tab:
+
+.. image:: img/extension_vscode.png
```
Comment (jdsika) [inline docs/testing.rst:91]: Lovely red arrows! ^^

**Review by jdsika — CHANGES_REQUESTED** [review]

**Comment by jdsika** [comment]
I would go for this extension packet: 
image

**Comment by jdsika** [comment]
Error: Tests are still creating files in the main directory:
image


Those should be created in the tests directory and be part of .gitignore

**Comment by vkresch** [comment]
@jdsika are you on ubuntu or windows?

**Comment by jdsika** [comment]
That was on Ubuntu, but I think I have the same problem on either OS

File: tests/__init__.py (hunk 2 above)
Comment (jdsika) [reply tests/__init__.py:4]: I resolved it

**Comment by jdsika** [comment]
The test fails locally as the dot.exe from graphviz is not found. But I don't want users to have to add it manually

**Comment by jdsika** [comment]
there is also a directory `logs/` created which should not be there. I think all temporary items from tests should be placed in one specific temp folder instead of being "spread all over the repo"

File: docs/fsm/graphviz/draw_state_diagrams.py (hunk 1 above)
Comment (vkresch) [reply docs/fsm/graphviz/draw_state_diagrams.py]: lol good point :). Resolved.

**Comment by jdsika** [comment]
@vkresch did you checkin this by intention? tests/tzkt/simulations/tz1NortRftucvAkD1J58L32EhSVrQEWJCEnB/calculations/201.csv

**Comment by vkresch** [comment]
@jdsika the user needs to install requirements.txt and requirements_developers.txt to make the test run successfully because we have a test which checks if all requirements are installed correctly. The artificats from pytest like logs/ and __pycache__/ are created from the smoke tests which simulate how you would normally run trd. Normally those artifacts would be created anyway. To avoid creating __pycache__/ folders you can use the -B flag when running tests or running the application.

File: docs/testing.rst (hunk 4 above)
Comment (vkresch) [reply docs/testing.rst:84]: thx :)

**Comment by vkresch** [comment]
Lint is failing due to Click incompability ([source](https://github.com/psf/black/issues/2964)).

**Comment by jdsika** [comment]
I am still getting this: FileNotFoundError: [Errno 2] No such file or directory: '/home/jdsika/tezos-reward-distributor/tests/unit/launch_state_diagram.png'

BUT the file IS there lol

**Comment by jdsika** [comment]
> @jdsika the user needs to install requirements.txt and requirements_developers.txt to make the test run successfully because we have a test which checks if all requirements are installed correctly. The artificats from pytest like logs/ and **pycache**/ are created from the smoke tests which simulate how you would normally run trd. Normally those artifacts would be created anyway. To avoid creating **pycache**/ folders you can use the -B flag when running tests or running the application.
> 
> I was able to run all tests sucessfully: image

The pycache folders are totally fine as they are also excluded in the .gitignore. I was more referring to the other folders which are not pycache.

**Review by jdsika — APPROVED** [review]

**Pull request merged**